proxy/http2/Http2Stream.cc

```diff
@@ -690,11 +691,14 @@ Http2Stream::send_response_body(bool call_update)
     // signal_write_event() will be called from `Http2ConnectionState::send_data_frames_depends_on_priority()`
     // when write_vio is consumed
   } else {
+    bool stream_deleted = false;
     SCOPED_MUTEX_LOCK(lock, parent->connection_state.mutex, this_ethread());
-    parent->connection_state.send_data_frames(this);
-    this->signal_write_event(call_update);
-    // XXX The call to signal_write_event can destroy/free the Http2Stream.
-    // Don't modify the Http2Stream after calling this method.
+    parent->connection_state.send_data_frames(this, stream_deleted);
```

Comment:
Maybe I'm missing something, but why not just change send_data_frames() to return a bool, rather than passing a reference and modify that?

e.g.
```
if (!parent->connection_state.send_data_frames(this)) {
    ....
}
```

Reply:
Yes, that would be equivalent.  I am ok either way.  I thought the parameter return would be more clear on the meaning.